### ISSUE TITLE
Build is failing on Python 2.7 dependency

### ISSUE THREAD

**Issue by garethbowen** [medic/cht-conf#562]
**Describe the issue**
GH Action is failing on setup with error...

```
Run actions/setup-python@v4
  with:
    python-version: 2.x
    architecture: x64
    check-latest: false
    token: ***
    update-environment: true
    allow-prereleases: false
Installed versions
  Warning: The support for python 2.7 will be removed on June 19. Related issue: https://github.com/actions/setup-python/issues/672
  Version 2.x was not found in the local cache
  Error: The version '2.x' with architecture 'x64' was not found for Ubuntu 20.04.
  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
```

Example build: https://github.com/medic/cht-conf/actions/runs/5323058024/jobs/9640333547

**Describe the improvement you'd like**
We need to either find a workaround or finally [upgrade to Python 3](https://github.com/medic/cht-core/issues/6537). Preferably the latter as Python 2 is EOL.

**Comment by garethbowen** [comment]
Set High priority because it's blocking the build right now.

**Comment by mrjones-plip** [comment]
@garethbowen  - the python 2.x upgrade [issue you linked to](https://github.com/medic/cht-core/issues/6537) is in CHT Core repo.  Reading over that ticket, it sounds like to do this upgrade to 2.x (vs finding a work around) we'd need to update Core as well as Conf - is that your understanding as well? 

I suspect doing the work around instead (ugh...terrible, I know) is [actually pretty easy](https://github.com/actions/runner-images/issues/7401#issuecomment-1571065641).

**Comment by garethbowen** [comment]
@mrjones-plip My interpretation is it's actually [pyxform that needs upgrading](https://github.com/medic/pyxform/issues/8) and everything else should just work. This needs to happen fairly soon anyway (it's a cascading set of dependencies which is holding up a whole range of things) but it's totally fine to do the workaround now and schedule the upgrade for another day...

**Comment by mrjones-plip** [comment]
Great thanks @garethbowen.  I'll defer to @jkuester on the best path forward here, as he opened that pyxform ticket you cited, and was in the details in the Core issue as well!

**Comment by yrimal** [comment]
I'm afraid if we opt to find the workaround this time as well, we'll not focus on upgrading pyxform and leaving all other dependencies behind. I totally understand leaving the build state in failing state is not the write thing to do, but upgrading pyxform to use python3 would make life a lot easier for every CHT developers as well.

**Comment by jkuester** [comment]
Well, I would very much like to get pyxform upgraded for a bunch of reasons!  Unfortunately, I do not think this Python upgrade is going to require pyxform changes.  

Medic's pyxform can be installed successfully on python3. So, I have opened [a PR](https://github.com/medic/cht-conf/pull/565) where I have just updated the cht-conf GitHub Action to use the default Python3 already installed on the `ubuntu` image.  The build passes just fine!
